internal/controller/numaresourcesscheduler_controller.go

```diff
@@ -171,13 +201,22 @@ func isDeploymentRunning(ctx context.Context, c client.Client, key nropv1.Namesp
 	return false, nil
 }
 
-func (r *NUMAResourcesSchedulerReconciler) computeSchedulerReplicas(schedSpec nropv1.NUMAResourcesSchedulerSpec) *int32 {
-	// the api validation/normalization layer must ensure this value is != nil
-	if *schedSpec.Replicas >= 0 { // 0 is legit value to disable the deployment
-		return schedSpec.Replicas
+func (r *NUMAResourcesSchedulerReconciler) computeSchedulerReplicas(ctx context.Context, schedSpec nropv1.NUMAResourcesSchedulerSpec) (*int32, error) {
+	// do not autodetect if explicitly set by the user
+	if schedSpec.Replicas != nil {
+		return schedSpec.Replicas, nil
+	}
+	nodeList := &corev1.NodeList{}
+	if err := r.Client.List(ctx, nodeList, &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			"node-role.kubernetes.io/control-plane": "",
+		}),
+	}); err != nil {
+		return schedSpec.Replicas, err
 	}
-	v := int32(r.AutodetectReplicas)
-	return &v
+	replicas := ptr.To(int32(len(nodeList.Items)))
```

Comment:
not needed, we can use a local variable and return its address. But not worth resubmitting just for this